container-storage-setup.sh

```diff
@@ -1109,6 +1109,86 @@ setup_extra_volume() {
   fi
 }
 
+# This is used only in compatibility mode. We are still using systemd
+# mount unit only for compatibility mode. Reason being that upon service
+# restart, we run into races and device might not yet be up when we try
+# to mount it. And we don't know if this is first time start and we need
+# to create volume or this is restart and we need to wait for device. So
+# continue to use systemd mount unit for compatibility mode.
+setup_systemd_mount_unit_compat() {
+  local filename
+  local vg=$1
+  local lv_name=$2
+  local mount_dir=$3
+  local unit_file_path
+
+  # filename must match the path ${mount_dir}.
+  # e.g if ${mount_dir} is /var/lib/containers
+  # then filename will be var-lib-containers.mount
+  filename=$(systemd_escaped_filename ${mount_dir})
+  unit_file_path="/etc/systemd/system/$filename"
+
+  # If unit file already exists, nothing to do.
+  [ -f "$unit_file_path" ] && return 0
+
+  cat <<EOF > "${unit_file_path}.tmp"
+[Unit]
+Description=Mount $lv_name on $mount_dir directory.
```

Comment:
Right now I am not expecting anybody to touch this file. As of now, we create xfs filesystem on extra volume and there are no knobs to configure it. So I am not expecting anyobdy to touch this file.

Down the line when we introduce any config variable to allow specifying filesystem, then we can specify filesystem type accordingly. 

But putting a warning that this file has been automatically generated by container-storage-setup is good anyway. Atleast system admin will know who created this file.